### PR TITLE
Hotfix/zl/2.0.6 2012.08.25 deletes

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Patch Code branch

### CODE REVIEW

File: src/yz_kv.erl [hunk 1]
```diff
@@ -396,6 +405,19 @@ cleanup(_, _) ->
 
 %% @private
 %%
+%% @doc Cleanup tombstones and siblings accordingly... cleanup/3
+%% TODO: make this work for allow_mult=true case
+-spec cleanup([doc()], bkey(), values()) -> [{bkey, bkey()}].
+cleanup([], _BKey, _Values) ->
+    [];
+cleanup([{doc, Fields}|T], BKey, [VH|VT]) ->
```
Comment (jonmeredith) [inline src/yz_kv.erl:413]: _DO NOT_ fix for patch, but I don't think we need Values at all - could just work on the Fields as is if a tombstone, VH will always be <<>>, so it's redundant.

Comment (zeeshanlakhani) [inline src/yz_kv.erl:413]: @jonmeredith good point.. actually was already doing this... but will do when i bring this over to the main PR.

**Comment by jonmeredith** [comment]
+1 c23e1f0

**Comment by zeeshanlakhani** [comment]
@borshop merge

**Pull request merged**